.changeset/dry-papers-warn.md

```diff
@@ -0,0 +1,31 @@
+---
+"@fluidframework/container-runtime": major
+"@fluidframework/container-runtime-definitions": minor
+"@fluidframework/datastore": major
+"@fluidframework/runtime-definitions": minor
+"@fluidframework/runtime-utils": minor
+"@fluid-private/test-end-to-end-tests": minor
+"@fluidframework/test-utils": minor
+---
+
+Implement ID compression for DataStores & DDSs
+
+
+# Key changes
+1. A new API IContainerRuntimeBase.generateDocumentUniqueId() is exposed. This API will opportunistically generate IDs in short format (non-negative numbers). If it can't achieve that, it will return a uuid string. It can be leveraged anywhere in container where container unique IDs are required. I.e. any place that uses uuid() and stores data in container is likely candidate to start leveraging this API.
```

Comment:
maybe worth mentioning somewhere that the UUIDs generated will also have very low entropy in groups and will compress well